sc-memory/sc-memory/sc_action.cpp

```diff
@@ -172,11 +172,13 @@ bool ScAction::InitiateAndWait(sc_uint32 waitTime_ms) noexcept(false)
 
   auto wait = std::shared_ptr<ScWaiterActionFinished>(new ScWaiterActionFinished(*m_context, *this));
   wait->SetOnWaitStartDelegate(
-      [this]()
+      [this, &expectedExecutionTimeInMilliseconds]()
       {
+        if (!m_context->IsElement(GetExpectedExecutionTimeInMilliseconds()))
+          SetExpectedExecutionTimeInMilliseconds(expectedExecutionTimeInMilliseconds);
```

Comment:
So here the getter is called twice: once in IsElement and once in the setter. Could you fix the code so that it is called once? I think you can unite this methods into one and use the united method here